### DISCUSSION TITLE
How to set X axis range to display N points

### DISCUSSION THREAD

**Discussion by BrumolaLuca** [patrykandpatrick/vico#802] [Questions] [answered]
### Question

I cant find out how to display like 120 points in my Line Cartesian Chart, im using AxisValueOverrider.fixed to set minX and maxX, but it seems the maximum of data display depends on the screen size..

Example :
`    val chartModelProducer = remember { CartesianChartModelProducer() }
    val axis =  AxisValueOverrider.fixed(
        minX = 1f,
        maxX = 60f,
        minY = 0f,
        maxY = 50f
    )

    val myChart = (
            rememberCartesianChart(
                rememberLineCartesianLayer(
                  axisValueOverrider = axis
                )
            ))

    LaunchedEffect(Unit) {
        while (true) {

            chartModelProducer.runTransaction {
                lineSeries{
                    series(x= x_vals, y = y_vals)
                }
            }

            y_vals.add( Random.nextFloat() * 3 + 30)
            y_vals.removeAt(0)

            delay(100) // Simulate data update every second
        }
    }

    CartesianChartHost(
        chart = myChart,
        diffAnimationSpec = tween(
            durationMillis = 0,
            delayMillis = 0,
            easing = LinearEasing
        ),
        modelProducer = chartModelProducer,
        modifier = Modifier
            .fillMaxSize()
            .padding(128.dp)
    )`
  

### Vico version(s)

Version: 2.0.0-alpha.23

### UI framework(s)

Jetpack Compose

**Comment by BrumolaLuca** [comment, marked as answer]
Resolved, the option was ZoomState : 

`   CartesianChartHost(
        chart = myChart,
        diffAnimationSpec = tween(
            durationMillis = 0,
            delayMillis = 0,
            easing = LinearEasing
        ),
        **zoomState =  rememberVicoZoomState(zoomEnabled = false, initialZoom = Zoom.x(100f)),**
        modelProducer = chartModelProducer,
        modifier = Modifier
            .fillMaxSize()
            .padding(128.dp)
    )`

**Reply by patrickmichalik** [reply]
Thanks for sharing!
